migrator/migrations/m_214_to_m_215_process_indicator_to_btree/migration_impl.go

```diff
@@ -0,0 +1,124 @@
+	deploymentColumn   = "deploymentid"
+
+	dropIndex   = "DROP INDEX if exists %s"
+	createIndex = "CREATE INDEX IF NOT EXISTS %s ON %s USING BTREE (%s)"
+	renameIndex = "ALTER INDEX IF EXISTS %s RENAME TO %s"
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+func migrate(database *types.Databases) error {
+	// We are simply changing the index type from hash to btree if the process_indicators
+	// indexes for deployment id and poduid are still hash.  There is at least one instance
+	// in the field where the indexes have already been moved to btree, we do not want to
+	// force that instance to remigrate.  So we will not simply drop and re-add, but
+	// verify index is hash, drop old index, re-add it as btree.
+	tx, err := database.PostgresDB.Begin(database.DBCtx)
+	if err != nil {
+		return err
+	}
+	ctx := postgres.ContextWithTx(database.DBCtx, tx)
+
+	// Purposefully doing this one at a time like this to be very specific on what we are doing.
+	err = migrateIndex(ctx, database.PostgresDB, deploymentIndex, deploymentColumn, tmpDeploymentIndex)
+	if err != nil {
+		return postgreshelper.WrapRollback(ctx, tx, err)
+	}
+
+	err = migrateIndex(ctx, database.PostgresDB, podIndex, podColumn, tmpPodIndex)
+	if err != nil {
+		return postgreshelper.WrapRollback(ctx, tx, err)
+	}
+
+	err = tx.Commit(ctx)
+	if err != nil {
+		return postgreshelper.WrapRollback(ctx, tx, errors.Wrapf(err, "unable to commit migration starting at %d", startSeqNum))
+	}
+
+	log.Infof("Process indicator index migration complete")
+	return nil
+}
+
+func migrateIndex(dbCtx context.Context, db postgres.DB, indexName string, indexColumn string, tmpIndexName string) error {
+	log.Infof("Migrating %s index %s on column %s to btree", tableName, indexName, indexColumn)
+	ctx, cancel := context.WithTimeout(dbCtx, types.DefaultMigrationTimeout)
+	defer cancel()
+
+	// Check if the hash index exists.  If it does not, we can fall through as either
+	// the index already exists in the desired type in which case we have nothing to do
+	// OR the index name does not exist at all in which case Gorm will take care of it when
+	// all schemas are applied at the end.
+	row := db.QueryRow(ctx, hashIndexQuery, tableName, indexName)
+	var exists bool
+	if err := row.Scan(&exists); err != nil {
+		return err
+	}
+
+	if !exists {
+		return nil
+	}
+
+	// To minimize time without an index and to care for the even of failure forcing a
+	// rollback which would recreate the hash index if failure occurred after the drop
+	// we are going to create a tmp new one, drop the old one, and then rename the
+	// temp one to minimize risk.
+
+	// create the new index concurrently
+	// Note: this is not really necessary because the indexes will be created when the
+	// schema updates are processed, HOWEVER, that will not create the indexes
+	// concurrently; we want to do them concurrently in this case for performance reasons.
+	createStatement := fmt.Sprintf(createIndex, tmpIndexName, tableName, indexColumn)
```

Comment:
The `createIndex` is actually not concurrent in this revision: `CREATE INDEX IF NOT EXISTS %s ON %s USING BTREE (%s)`. Therefore, the comment is misleading.

WDYT about the idea to:
1. Create both new temp indexes concurrently with `IF NOT EXIST`.
2. Wait for that to complete. IDK how yet.
3. Open a transaction and do both drop+renames in it.

?